cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/client/AzureClient.java

```diff
@@ -513,6 +514,13 @@ public Mono<Void> deallocateVirtualMachineAsync(String resourceGroup, String vmN
         return handleException(() -> azure.virtualMachines().deallocateAsync(resourceGroup, vmName));
     }
 
+    public Mono<Void> deallocateVirtualMachineAsync(String resourceGroup, String vmName, Long timeboundInMs) {
+        if (timeboundInMs == null) {
+            return deallocateVirtualMachineAsync(resourceGroup, vmName);
+        }
+        return handleException(() -> azure.virtualMachines().deallocateAsync(resourceGroup, vmName)).timeout(Duration.ofMillis(timeboundInMs));
```

Comment:
The timeout here is on the handleException return value.
On the startAsync - it's on the invocation of the startAsync call (as against the handleException)

Which one is it supposed to be ?

Reply:
It should be on the function call.
So, I have updated this